flagd/pkg/service/flag-sync/sync_service.go

```diff
@@ -0,0 +1,164 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"slices"
+	"time"
+
+	"buf.build/gen/go/open-feature/flagd/grpc/go/flagd/sync/v1/syncv1grpc"
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/store"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+)
+
+type ISyncService interface {
+	// Start the sync service
+	Start(context.Context) error
+
+	// Emit updates for sync listeners
+	Emit(isResync bool, source string)
+}
+
+type SvcConfigurations struct {
+	Logger  *logger.Logger
+	Port    uint16
+	Sources []string
+	Store   *store.Flags
+}
+
+type Service struct {
+	listener net.Listener
+	logger   *logger.Logger
+	mux      *Multiplexer
+	server   *grpc.Server
+
+	startupTracker syncTracker
+}
+
+func NewSyncService(cfg SvcConfigurations) (*Service, error) {
+	l := cfg.Logger
+	mux, err := NewMux(cfg.Store, cfg.Sources)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing multiplexer: %w", err)
+	}
+
+	server := grpc.NewServer()
+	syncv1grpc.RegisterFlagSyncServiceServer(server, &syncHandler{
+		mux: mux,
+		log: l,
+	})
+
+	l.Info(fmt.Sprintf("starting flag sync service on port %d", cfg.Port))
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", cfg.Port))
+	if err != nil {
+		return nil, fmt.Errorf("error creating listener: %w", err)
+	}
+
+	return &Service{
+		listener: listener,
+		logger:   l,
+		mux:      mux,
+		server:   server,
+		startupTracker: syncTracker{
+			sources:  slices.Clone(cfg.Sources),
+			doneChan: make(chan interface{}),
+		},
+	}, nil
+}
+
+func (s *Service) Start(ctx context.Context) error {
+	// derive errgroup so we track ctx for exit as well as startup errors
+	g, lCtx := errgroup.WithContext(ctx)
+
+	g.Go(func() error {
+		// delay server start until we see all syncs from known sync sources OR timeout
+		select {
+		case <-time.After(5 * time.Second):
+			s.logger.Warn("timeout waiting for all sync sources to complete their initial sync. " +
```

Comment:
I am not a native speaker but this sounds more readable 🤔 
```suggestion
			s.logger.Warn("finish waiting for all sync sources to complete their initial sync. " +
```

Reply:
I tried to improve this log. Here the meaning should include a timeout as this is a timeout waiting for all sync sources to complete.